app/views/categories/_categories.html.haml

```diff
@@ -0,0 +1,18 @@
+%ul.category-summary
+  - categories.each do |category|
+    - if category.has_children?
+      - count = category.children.inject(0) do |memo, child|
```

Comment:
> So I want to reduce loop here.

Hmm, it seems we can't reduce loop at any rate.

In this case, `sum` is more explanatory than `inject` I think.

---

`sum` accepts block:

```ruby
count = category.children.sum {|child| child.posts.published.count }
```